keras_eval/data_generators.py

```diff
@@ -0,0 +1,328 @@
+import os
+import numpy as np
+import PIL
+
+from keras_preprocessing.image import ImageDataGenerator, DirectoryIterator, get_keras_submodule, array_to_img, \
+    load_img, img_to_array
+
+backend = get_keras_submodule('backend')
+
+
+class AugmentedDirectoryIterator(DirectoryIterator):
+    '''
+
+    AugmentedDirectoryIterator inherits from DirectoryIterator:
+    (https://github.com/keras-team/keras-preprocessing/blob/master/keras_preprocessing/image.py#L1811)
+
+    Added functionality of doing multiple crops following the paper Going Deeper with Convolutions
+    (https://arxiv.org/pdf/1409.4842.pdf) and allowing the use of performing transforms on the crops.
+
+    Has the addition of data_augmentation as an argument. It is a dictionary consisting of 3 elements,
+    'scale_sizes' could be 'default' - similar to Google's paper or a list of sizes. Each scaled image then will
+    be cropped in three square parts, 'transforms' is a list of transforms to apply to these crops in addition to not
+    applying any transform ('horizontal_flip', 'vertical_flip', 'rotate_90', 'rotate_180', 'rotate_270' are
+    supported now) and 'crop_original' that allows to center crop the original image prior do the rest of transforms
+    and scale + croppings.
+
+    E.g. data_augmentation={'scale_sizes':'default', 'transforms':['horizontal_flip', 'rotate_180'],
+    'crop_original':'center_crop'}
+
+    '''
+
+    def __init__(self, directory, image_data_generator, data_augmentation,
+                 target_size=(299, 299), color_mode='rgb',
+                 classes=None, class_mode='categorical',
+                 batch_size=32, shuffle=True, seed=None,
+                 data_format='channels_last',
+                 save_to_dir=None, save_prefix='', save_format='png',
+                 follow_links=False,
+                 subset=None,
+                 interpolation='nearest',
+                 ):
```

Comment:
In future iterations, maybe we could provide the definitions of these attributes for a better understanding.

Reply:
These arguments are already defined in the keras superclass